File: Components/Tex_ture/PI_NRemoteImage/Examples/Example/PI_NRemoteImage/DegradedViewController.m
//
//  DegradedViewController.m
//  PI_NRemoteImage
//
//  Created by Garrett Moon on 7/14/15.
//  Copyright (c) 2015 Garrett Moon. All rights reserved.
//

#import "DegradedViewController.h"

#import <PI_NRemoteImage/PI_NImageView+PI_NRemoteImage.h>

@interface DegradedViewController ()

@end

@implementation DegradedViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    [[PI_NRemoteImageManager sharedImageManager] setShouldUpgradeLowQualityImages:YES completion:nil];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    [self.imageView pin_setImageFromURLs:@[[NSURL URLWithString:@"https://placekitten.com/101/101"],
                                           [NSURL URLWithString:@"https://placekitten.com/401/401"],
                                           [NSURL URLWithString:@"https://placekitten.com/801/801"]]];
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: Components/Tex_ture/PI_NRemoteImage/Examples/Example-tvOS/PI_NRemoteImage.tvOSExample/DegradedViewController.swift
//
//  DegradedViewController.swift
//  PI_NRemoteImage.tvOSExample
//
//  Created by Isaac Overacker on 2/6/16.
//
//

import UIKit
import PI_NRemoteImage

class DegradedViewController: UIViewController {

    @IBOutlet weak var imageView: UIImageView!

    override func viewDidLoad() {
        super.viewDidLoad()
        PI_NRemoteImageManager.shared().setShouldUpgradeLowQualityImages(true, completion: nil)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)

        imageView.pin_setImage(from: [
            URL(string: "https://placekitten.com/101/101")!,
            URL(string: "https://placekitten.com/401/401")!,
            URL(string: "https://placekitten.com/801/801")!
        ])
    }
}
